gnosis/safe-react
# Don't block UI when on a wrong chain

## What it solves
Resolves #2924

## How this PR fixes it
Resolves the 3-click problem by not showing the backdrop and a network switch popup when connected to a "wrong" chain.
However, the Safe will be read-only.

A note: a lot of transaction-related logic depends on Web3 being initialized before a tx-initiating component has been rendered. We're not using hooks enough in this area.

## How to test it
* Connect to Polygon
* Go to a Rinkeby Safe
* Safe should be read-only
* Switch to Rinkeby
* Safe should become active
* Switch to xDai
* Safe should become readonly

## Screenshot
<img width="996" alt="Screenshot 2021-11-16 at 13 12 25" src="https://user-images.githubusercontent.com/381895/141983289-0175c066-71b5-43e2-9445-71c23e974cf3.png">
